packages/actor-extract-links-solid-type-index/test/ActorExtractLinksSolidTypeIndex-test.ts

```diff
@@ -315,6 +316,93 @@ describe('ActorExtractLinksSolidTypeIndex', () => {
       expect(mediatorDereferenceRdf.mediate).toHaveBeenCalledWith({ url: 'ex:index1', context });
     });
 
+    it('should run on a stream with type index predicates without class in query', async() => {
+      input = stream([
+        quad('ex:s1', 'ex:typeIndex1', 'ex:index1'),
+        quad('ex:s3', 'ex:px', 'ex:o3', 'ex:gx'),
+        quad('ex:s4', 'ex:p', 'ex:o4', 'ex:g'),
+        quad('ex:s5', 'ex:p', 'ex:o5', 'ex:gx'),
+      ]);
+      (<any> actor).queryEngine = {
+        queryBindings: jest.fn(async() => ({
+          toArray: async() => [
+            BF.fromRecord({ instance: DF.namedNode('ex:file4'),
+              class: DF.namedNode('ex:class4'),
+              domain: DF.namedNode('ex:class4'),
+              s: DF.namedNode('ex:predicate1') }),
+          ],
+        })),
+      };
+      context = new ActionContext({
+        [KeysInitQuery.query.name]: AF.createPattern(
+          DF.variable('s'),
+          DF.namedNode('ex:predicate1'),
+          DF.namedNode('ex:blabla'),
+        ),
+        [KeysQueryOperation.operation.name]: AF.createPattern(
+          DF.variable('s'),
+          DF.namedNode('ex:predicate1'),
+          DF.namedNode('ex:bla'),
+        ),
+      });
+      await expect(actor.run({ url: '', metadata: input, requestTime: 0, context })).resolves
+        .toEqual({
+          links: [
+            {
+              url: 'ex:file4',
+            },
+          ],
+        });
+      expect(mediatorDereferenceRdf.mediate).toHaveBeenCalledTimes(1);
+      expect(mediatorDereferenceRdf.mediate).toHaveBeenCalledWith({ url: 'ex:index1', context });
+    });
+
+    it('should run on a stream with multiple type index predicates without class in query', async() => {
```

Comment:
Could we also have a test that builds upon this one, and exposes type index entries for other predicates (like predicate2 and predicate3) as well?
The goal is to check that these entries are ignored by the actor, since we're only using predicate1 in the query.